x-pack/plugin/core/src/main/resources/logs-mappings.json

```diff
@@ -2,6 +2,25 @@
   "template": {
     "mappings": {
       "dynamic_templates": [
+        {
+          "match_ip": {
```

Comment:
@jpountz With the recent development in ES, I started to ask myself if these should actually all be dynamic mapping for runtime fields? It would change the PR a bit as I think in this case we should bring back the global `message` field but everything else should match as runtime.

Reply:
In addition, having runtime fields would allow us to potentially also have runtime fields for things like `*_time` or `*_date` so these would work and still not be indexed. It would also allow an "easy" fix in case the detection was not correct (I assume).

Reply:
Dynamic mappings are going to be a security issue.  Are there other ways to do this?

Reply:
@ruflin Can you dig a bit more into why you're thinking about using runtime fields?

If you are worried of failing to index some documents, then maybe we should add `ignore_malformed:true` to these `ip` mappings?

If you would like to allow users to change their minds, then either runtime fields or concrete fields is fine since we allow runtime fields to shadow the definition of concrete fields, so users would have the leisure of configuring their field differently via a runtime field on existing indices if they wanted to regardless of the decision we would make here.

@scunningham There is no other way to do it. My understanding of the parallel thread on #68414 is that we would still allow dynamic mappings for trusted endpoints, so I believe that fine-tuning dynamic templates in these base templates still provides value.

Reply:
Lets assume we have a document with several `*.ip` fields inside. By default currently these are mapped to keywords which is not correct. So if someone wants to query for the `*.ip` fields, the result is unexpected because of the wrong type. In the ideal case, these fields would not be indexed at all because my assumption is, in most cases these are not used. And if they are used, a more specific dataset with a template will likely exist and have the mappings for it.

The above goes further: We currently index all the things as keyword. Instead we should only index a few fields and not index all the others and instead use runtime fields.

Reply:
> And if they are used, a more specific dataset with a template will likely exist and have the mappings for it.

I think this holds true for our integrations, but we also need to take care of users who are onboarding new sources of data, e.g. because it's custom or because we don't have an integration for it (yet)?

My understanding is that `host.ip` should generally not be used, so we could make this one runtime. But the source IP field of a `nginx.access` dataset should be indexed in order to allow users to run cardinality aggregations on this field (how many users per day) or compare IPv4 traffic vs IPv6 traffic? In general it feels hard to me to make the assumption about the data that because it is an IP field, then generally it is not useful, even if that's the case most of the time in ECS.

What about mapping `host.ip` as a runtime field and keeping that template for other fields whose name ends with `.ip` and map them as indexed `ip` fields?

Reply:
I checked with @tsg and the Security solution leverages these IP fields in a dashboard, so we should keep them indexed, at least for now.

Reply:
@tsg What we discuss here is the default. Are the queries you run in the context of installed packages? If yes, these could still have a more specific definition that makes it index. The part here is only the global default if no other template exists.